src/controllers/authController.js

```diff
@@ -65,8 +67,43 @@ const logout = async (req, res) => {
   res.status(StatusCodes.OK).json({ msg: 'user logged out!' });
 };
 
+//forgotPassword endpoint
+const forgotPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw new CustomError.BadRequestError('Please provide an email');
+  }
+
+  const user = await User.findOne({ email });
+
+  if (!user) {
+    throw new CustomError.NotFoundError('User not found');
+  }
+
+  // Generate a random token
+  const resetToken = crypto.randomBytes(20).toString('hex');
+  const resetExpires = Date.now() + 15 * 60 * 1000; // 15 minutes from now
+
+  // Update user's reset token and expiration date
+  user.passwordResetToken = resetToken;
+  user.passwordResetExpiresOn = new Date(resetExpires);
+  await user.save();
+
+  // Send reset password email
+  const emailTemplate = 'forgotPassword'; // to adjust this based on our email template name
+  const recipient = user.email;
+  const localVariables = {
+    resetToken,
+  };
+  await emailSender.send(emailTemplate, recipient, localVariables);
```

Comment:
This function doesn't exist in your `emailSender`. Should be:

```suggestion
  await emailSender.sendForgotPasswordEmail(emailTemplate, recipient, localVariables);
```